webui/react/src/hooks/useModal/useModalExperimentCreate.tsx

```diff
@@ -215,15 +221,38 @@ const useModalExperimentCreate = (props?: Props): ModalHooks => {
     const error = modalState.error || modalState.configError;
     if (error) throw new Error(error);
 
-    let configString;
+    let fullConfig;
     if (!modalState.isAdvancedMode) {
       await formRef.current?.validateFields();
-      configString = getConfigFromForm(modalState.config);
+      const publicConfig = modalState.config;
+      if (registryCredentials) {
+        const { environment, ...restConfig } = publicConfig;
+        fullConfig = {
+          environment: { registry_auth: registryCredentials, ...environment },
+          ...restConfig,
+        };
+      } else {
+        fullConfig = publicConfig;
+      }
+
     } else {
-      configString = modalState.configString;
+
+      const userConfig = (yaml.load(modalState.configString) || {}) as RawJson;
+
+      if(!userConfig?.environment?.registry_auth && registryCredentials) {
+        const { environment, ...restConfig } = userConfig;
+        fullConfig = {
+          environment: { registry_auth: registryCredentials, ...environment },
+          ...restConfig,
+        };
+      } else {
+        fullConfig = userConfig;
+      }
+
```

Comment:
**super nit suggestion:** this is pretty easy to read so feel free to ignore this. The logic can probably simplify a little, assuming the `SimpleMode` can follow the same exact logic as the `AdvancedMode` where you check to see if `environment?.registry_auth` is NOT defined and `registryCredentials` is to spot update `config > environment > registry_auth`.

Reply:
did something similar to simplify 👍